packages/vscode/src/plugins/nextjs/index.ts

```diff
@@ -0,0 +1,83 @@
+import { Project, ProjectOptions } from 'ts-morph'
+import { window } from 'vscode'
+import { packageJsonIncludes } from '../../helpers/workspace'
+import { PrismaVSCodePlugin } from '../types'
+import { addMissingImports } from './addImports'
+import { addJSDoc } from './addJSDoc'
+import { addTypesToPage } from './addTypesToPage'
+import { getUsedNextFunctions, isJS } from './utils'
+
+interface Options {
+  tsProjectOptions?: ProjectOptions
+  /**
+   * default:  "./pages/**\/*.{ts,tsx,js,jsx}"
+   */
+  customPagesGlob?: string
+  /**
+   * default:  true
+   */
+  save: boolean
+}
+
+export class NextTypes {
+  project: Project
+  options: Options | undefined
+  constructor(options?: Options) {
+    this.project = new Project({
+      skipFileDependencyResolution: true,
+      skipLoadingLibFiles: false,
+      ...options?.tsProjectOptions,
+      compilerOptions: {
+        allowJs: true,
+        ...options?.tsProjectOptions?.compilerOptions,
+      },
+    })
+    this.options = options
+    // const pagesGlob = options?.customPagesGlob || './pages/**/*.{ts,tsx,js,jsx}'
+    // const globs = join(projectDir, pagesGlob)
+    // Add source files
+  }
+  async run(filePath: string): Promise<void> {
+    // Read more: https://ts-morph.com/setup/
+    const sourceFile = this.project.addSourceFileAtPath(filePath)
+
+    const foundFunctions = getUsedNextFunctions(sourceFile)
+    if (isJS(sourceFile)) {
+      addJSDoc(sourceFile, foundFunctions)
+    } else {
+      addMissingImports(sourceFile, foundFunctions)
+      addTypesToPage(sourceFile, foundFunctions)
+    }
+    if (this.options?.save) {
+      await this.project.save()
+    }
+    this.project.removeSourceFile(sourceFile)
+  }
+}
+
+const nextTypes = new NextTypes({ save: true })
+
+const plugin: PrismaVSCodePlugin = {
+  name: 'nextjs',
+  enabled: () => packageJsonIncludes('next'),
+  commands: [
+    {
+      id: 'prisma.plugin.nextjs.addTypes',
+      action: async () => {
+        const filename = window.activeTextEditor?.document.fileName
+        if (filename && filename.includes('pages')) {
```

Comment:
Is this the wanted behaviour? Does the filename in such a case always include `pages`?

Reply:
Yes